### PR TITLE
[RW-5640][risk=low] Adding UI to add microarray codegen

### PR DESCRIPTION
**Pull request by ericsong**
Description:
Adding UI flow to add microarray code gen
image

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI + API server with `yarn test-local`
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/pages/data/data-set/new-dataset-modal.spec.tsx [hunk 1]
```diff
@@ -11,7 +11,10 @@ import {
 import {waitOneTickAndUpdate} from 'testing/react-test-helpers';
 import {DataSetApiStub} from 'testing/stubs/data-set-api-stub';
 import {WorkspacesApiStub} from 'testing/stubs/workspaces-api-stub';
+import {DataSetExportRequest} from '../../../../generated/fetch';
```
Comment (calbach) [inline ui/src/app/pages/data/data-set/new-dataset-modal.spec.tsx:14]: please make relative to application, not the current file (next line also violates this, but is less offensive and existing issue)

File: ui/src/app/pages/data/data-set/new-dataset-modal.spec.tsx [hunk 2]
```diff
@@ -11,7 +11,10 @@ import {
...
 import {NewDataSetModal} from './new-dataset-modal';
+import GenomicsAnalysisToolEnum = DataSetExportRequest.GenomicsAnalysisToolEnum;
```
Comment (calbach) [inline ui/src/app/pages/data/data-set/new-dataset-modal.spec.tsx:16]: interesting - I actually haven't seen this pattern before but it seems pretty nice

File: ui/src/app/pages/data/data-set/new-dataset-modal.tsx [hunk 3]
```diff
@@ -1,30 +1,36 @@
...
 import {TooltipTrigger} from 'app/components/popups';
 import {SpinnerOverlay} from 'app/components/spinners';
+import {TextColumn} from 'app/components/text-column';
 import {appendNotebookFileSuffix} from 'app/pages/analysis/util';
+
+import {dataSetApi, workspacesApi} from 'app/services/swagger-fetch-clients';
 import colors from 'app/styles/colors';
-import {summarizeErrors} from 'app/utils';
+import {
+  summarizeErrors
```
Comment (calbach) [inline ui/src/app/pages/data/data-set/new-dataset-modal.tsx:14]: nit: would revert these newlines

File: ui/src/app/pages/data/data-set/new-dataset-modal.spec.tsx [hunk 4]
```diff
@@ -139,17 +148,103 @@ describe('NewDataSetModal', () => {
...
+      includesAllParticipants: false,
+      description: '',
+      conceptSetIds: [],
+      cohortIds: [],
+      domainValuePairs: [],
+      prePackagedConceptSet: PrePackagedConceptSetEnum.NONE
+    };
+
+    wrapper.find('[data-test-id="data-set-name-input"]')
+    .first().simulate('change', {target: {value: nameStub}});
```
Comment (calbach) [inline ui/src/app/pages/data/data-set/new-dataset-modal.spec.tsx:223]: nit: indent, here and below

**Review by calbach — APPROVED** [review]
Nice! Looks pretty clean - I couldn't find any real issues.

File: ui/src/app/pages/data/data-set/new-dataset-modal.tsx [hunk 5]
```diff
@@ -339,6 +351,42 @@ class NewDataSetModal extends React.Component<Props, State> {
                 </label>
               )}
           </React.Fragment>}
+          {this.props.displayMicroarrayOptions && this.state.kernelType === KernelTypeEnum.Python &&
```
Comment (calbach) [inline ui/src/app/pages/data/data-set/new-dataset-modal.tsx:354]: OK, forgot to flag one issue that I believe exists here: should ensure that the following doesn't 400 (leave it up to you the best method for doing so):

- check microarray
- switch to R
- export

Comment (calbach) [reply ui/src/app/pages/data/data-set/new-dataset-modal.tsx:354]: (right now, pretty sure the includeMicroarrayData will still be `true`, resulting in an export failure to R)

Comment (ericsong) [reply ui/src/app/pages/data/data-set/new-dataset-modal.tsx:354]: good find. fixed!

**Comment by ericsong** [comment]
puppeteer failed but its also failing on master so I'm going to go ahead with a merge

**Pull request merged**